test/abstractarray.jl

```diff
@@ -1286,20 +1286,24 @@ end
     ax=axes(a,1)
     @test a==a[ax]
     a=reshape(collect(1:100),(10,10))
-    @test a==a[ax,ax]
+    @test a[:,1]==a[ax,1]
     a=collect(1:10)
     ax=Base.OneTo(8)
     @test a[1:8]==a[ax]
     a=reshape(collect(1:100),(10,10))
-    ay=Base.OneTo(7)
-    @test a[1:8,1:7]==a[ax,ay]
+    @test a[1:8,1]==a[ax,1]
     ax=UnitRange(1,10)
     @test a[ax]==collect(1:10)
-    @test a[ax,ax]==reshape(collect(1:100),(10,10))
+    @test a[ax,1]==collect(1:10)
```

Comment:
I guess you don't need to collect on the right hand side, the equality check should work an elementwise comparison anyway. This might save some allocation (pretty small though in this case)

Reply:
Changes did as asked.